contracts/transfer/tests/transfer.rs

```diff
@@ -553,3 +563,297 @@ fn phoenix_deposit_and_withdraw() {
+
+    let rng = &mut StdRng::seed_from_u64(0xfeeb);
+
+    let phoenix_sk = SecretKey::random(rng);
+    let phoenix_vk = ViewKey::from(&phoenix_sk);
+    let phoenix_pk = PublicKey::from(&phoenix_sk);
+
+    let moonlight_sk = BlsSecretKey::random(rng);
+    let moonlight_pk = BlsPublicKey::from(&moonlight_sk);
+
+    let vm = &mut rusk_abi::new_ephemeral_vm()
+        .expect("Creating ephemeral VM should work");
+    let mut session = &mut instantiate(rng, vm, &phoenix_pk, &moonlight_pk);
+
+    let swapper_account = account(&mut session, &moonlight_pk)
+        .expect("Getting account should succeed");
+    let nonce = swapper_account.nonce + 1;
+
+    assert_eq!(
+        swapper_account.balance, MOONLIGHT_GENESIS_VALUE,
+        "The swapper's account should have the genesis value"
+    );
+
+    let leaves = leaves_from_height(session, 1)
+        .expect("getting the notes should succeed");
+    let notes = filter_notes_owned_by(
+        phoenix_vk,
+        leaves.into_iter().map(|leaf| leaf.note),
+    );
+
+    assert_eq!(notes.len(), 0, "There should be no notes at this height");
+
+    let address =
+        phoenix_pk.gen_stealth_address(&JubJubScalar::random(&mut *rng));
+    let note_sk = phoenix_sk.gen_note_sk(&address);
+
+    let convert = Withdraw::new(
+        rng,
+        &note_sk,
+        TRANSFER_CONTRACT.to_bytes(),
+        SWAP_VALUE,
+        WithdrawReceiver::Phoenix(address),
+        WithdrawReplayToken::Moonlight(nonce),
+    );
+
+    let contract_call = ContractCall {
+        contract: TRANSFER_CONTRACT.to_bytes(),
+        fn_name: String::from("convert"),
+        fn_args: rkyv::to_bytes::<_, 1024>(&convert)
+            .expect("should serialize conversion correctly")
+            .to_vec(),
+    };
+
+    let tx = create_moonlight_transaction(
+        &moonlight_sk,
+        None,
+        0,
+        SWAP_VALUE,
+        GAS_LIMIT,
+        LUX,
+        nonce,
+        Some(contract_call),
+    );
+
+    let gas_spent = execute(&mut session, tx)
+        .expect("Executing transaction should succeed")
+        .gas_spent;
+    update_root(session).expect("Updating the root should succeed");
+
+    println!("CONVERT moonlight to phoenix: {} gas", gas_spent);
+
+    let swapper_account = account(&mut session, &moonlight_pk)
+        .expect("Getting account should succeed");
+
+    assert_eq!(
+        swapper_account.balance,
+        MOONLIGHT_GENESIS_VALUE - gas_spent - SWAP_VALUE,
+        "The swapper's account should have had the swap value subtracted along with gas spent"
+    );
+
+    let leaves = leaves_from_height(session, 1)
+        .expect("getting the notes should succeed");
+    let notes = filter_notes_owned_by(
+        phoenix_vk,
+        leaves.into_iter().map(|leaf| leaf.note),
+    );
+    let notes_value = owned_notes_value(phoenix_vk, &notes);
+
+    assert_eq!(notes.len(), 1, "A new note should have been created");
+    assert_eq!(
+        notes_value, SWAP_VALUE,
```

Comment:
Isn't the phoenix-pk instantiated with a `PHOENIX_GENESIS_VALUE` just like moonlight? If so this should be `PHOENIX_GENESIS_VALUE + SWAP_VALUE`, shouldn't it?

Reply:
We only get the notes from a given block height, meaning we're not getting the genesis ones.